lib/src/Pos/Node/API.hs

```diff
@@ -0,0 +1,552 @@
+        svNumber <- o .: "version"
+        pure $ V1 Core.SoftwareVersion{..}
+
+instance ToSchema (V1 Core.SoftwareVersion) where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "V1SoftwareVersion") $ mempty
+            & type_ .~ SwaggerObject
+            & properties .~ (mempty
+                & at "applicationName" ?~ Inline (toSchema (Proxy @Text))
+                & at "version" ?~ Inline (toSchema (Proxy @Word32))
+            )
+            & required .~ ["applicationName", "version"]
+
+instance Arbitrary (V1 Core.SoftwareVersion) where
+    arbitrary = fmap V1 arbitrary
+
+instance Buildable Version where
+    build v = bprint shown v
+
+instance Buildable a => Buildable (V1 a) where
+    build (V1 x) = bprint build x
+
+#if !(MIN_VERSION_swagger2(2,2,2))
+-- See note [Version Orphan]
+instance ToSchema Version where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "Version") $ mempty
+            & type_ .~ SwaggerString
+
+-- Note [Version Orphan]
+-- I have opened a PR to add an instance of 'Version' to the swagger2
+-- library. When the PR is merged, we can delete the instance here and remove the warning from the file.
+-- PR: https://github.com/GetShopTV/swagger2/pull/152
+#endif
+
+-- | The @static@ settings for this wallet node. In particular, we could group
+-- here protocol-related settings like the slot duration, the transaction max size,
+-- the current software version running on the node, etc.
+data NodeSettings = NodeSettings
+    { setSlotDuration   :: !SlotDuration
+    , setSoftwareInfo   :: !(V1 Core.SoftwareVersion)
+    , setProjectVersion :: !Version
+    , setGitRevision    :: !Text
+    } deriving (Show, Eq, Generic)
+
+deriveJSON Serokell.defaultOptions ''NodeSettings
+
+instance ToSchema NodeSettings where
+  declareNamedSchema =
+    genericSchemaDroppingPrefix "set" (\(--^) props -> props
+      & ("slotDuration"   --^ "Duration of a slot.")
+      & ("softwareInfo"   --^ "Various pieces of information about the current software.")
+      & ("projectVersion" --^ "Current project's version.")
+      & ("gitRevision"    --^ "Git revision of this deployment.")
+    )
+
+instance Arbitrary NodeSettings where
+    arbitrary = NodeSettings <$> arbitrary
+                             <*> arbitrary
+                             <*> arbitrary
+                             <*> pure "0e1c9322a"
+
+instance Example NodeSettings
+deriveSafeBuildable ''NodeSettings
+instance BuildableSafeGen NodeSettings where
+    buildSafeGen _ NodeSettings{..} = bprint ("{"
+        %" slotDuration="%build
+        %" softwareInfo="%build
+        %" projectRevision="%build
+        %" gitRevision="%build
+        %" }")
+        setSlotDuration
+        setSoftwareInfo
+        setProjectVersion
+        setGitRevision
+
+
+-- The API definition is down here for now due to TH staging restrictions. Will
+-- relocate other stuff into it's own module when the extraction is complete.
+type API =
```

Comment:
So this is the entire node-related API as served by the wallet in `v1`. I might split it up into eg `node-info`, `node-settings`, and `node-update` types so that the `wallet` package can implement them in a different way wrt SErvant definitions. Otherwise, I think this is mostly fine.